src/range.ts

```diff
@@ -3,27 +3,51 @@ import { IterableProtocol, RangeIterator } from './commons/Iterators';
 
 function range(
     start: number,
-    end: number,
-    step: number | Function
+    end?: number,
+    step?: number | Function
 ): IterableProtocol {
-    if (start === end) {
-        throw new ArgumentError('The start and end parameter is same.');
-    } else if (
-        typeof start !== 'number' ||
-        typeof end !== 'number' ||
-        (typeof step !== 'number' && typeof step !== 'function')
-    ) {
+    if (typeof end === 'undefined') {
+        end = start;
+        start = 0;
+    }
+
+    if (typeof step === 'undefined') {
+        if (start < end)
+            step = 1;
+        else
+            step = -1;
+    }
+
+    if (validateSameRage(start, end)) {
+        throw new ArgumentError('The range parameter is same.');
+    } else if (validateInputTypes(start, end, step)) {
         throw new ArgumentError('Please check arguments type.');
     }
 
     let stepFunction: any = step;
+
     if (typeof step === 'number') {
-        stepFunction = (value: number) => {
+        stepFunction = (value: any) => {
```

Comment:
This `any` here... looks like a little bit weird. Could we fix this to number?

Reply:
The number is occur TS compile error. I consider another thing without any. Thank you